### PR TITLE
Make Remote Publication a static setting

### PR DESCRIPTION
**Pull request by shiv0408**
### Description
Made the Remote publication setting a dynamic setting instead of a feature flag. Also, whenever the setting is toggled, we send the full state to all the nodes/remote store.  

### Check List
- [ ] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by soosinha — APPROVED** [review]

**Review by shwetathareja — APPROVED** [review]

**Comment by shwetathareja** [comment]
@shiv0408 please add change log also.

**Comment by shwetathareja** [comment]
@shiv0408 Remote state test failed
[org.opensearch.gateway.remote.RemoteClusterStateServiceTests.testWriteIncrementalMetadataSuccessWhenPublicationEnabled](https://build.ci.opensearch.org/job/gradle-check/46966/testReport/junit/org.opensearch.gateway.remote/RemoteClusterStateServiceTests/testWriteIncrementalMetadataSuccessWhenPublicationEnabled/)
    [org.opensearch.gateway.remote.RemoteClusterStateServiceTests.testWriteFullMetadataSuccessPublicationEnabled](https://build.ci.opensearch.org/job/gradle-check/46966/testReport/junit/org.opensearch.gateway.remote/RemoteClusterStateServiceTests/testWriteFullMetadataSuccessPublicationEnabled/)

File: server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java [hunk 1]
```diff
@@ -123,6 +121,18 @@ public class RemoteClusterStateService implements Closeable {
 
     private static final Logger logger = LogManager.getLogger(RemoteClusterStateService.class);
 
+    /**
+     * Gates the functionality of remote publication.
+     */
+    public static final String REMOTE_PUBLICATION_SETTING_KEY = "opensearch.feature.remote_store.publication.enabled";
```
Comment (soosinha) [inline server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java:127]: Can you change the setting name to "cluster.remote_store.publication.enabled" ?

File: server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java [hunk 2]
```diff
@@ -749,7 +749,7 @@ public void testWriteIncrementalMetadataSuccess() throws IOException {
 
     public void testWriteIncrementalMetadataSuccessWhenPublicationEnabled() throws IOException {
         publicationEnabled = true;
-        Settings nodeSettings = Settings.builder().put(REMOTE_PUBLICATION_EXPERIMENTAL, publicationEnabled).build();
+        Settings nodeSettings = Settings.builder().put(REMOTE_PUBLICATION_SETTING_KEY, publicationEnabled).build();
         FeatureFlags.initializeFeatureFlags(nodeSettings);
```
Comment (soosinha) [inline server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java:753]: This is not required instead it should be added in settings during setup

File: server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java [hunk 3]
```diff
@@ -371,7 +371,7 @@ public void testWriteFullMetadataSuccess() throws IOException {
     public void testWriteFullMetadataSuccessPublicationEnabled() throws IOException {
         // TODO Make the publication flag parameterized
         publicationEnabled = true;
-        Settings nodeSettings = Settings.builder().put(REMOTE_PUBLICATION_EXPERIMENTAL, publicationEnabled).build();
+        Settings nodeSettings = Settings.builder().put(REMOTE_PUBLICATION_SETTING_KEY, publicationEnabled).build();
         FeatureFlags.initializeFeatureFlags(nodeSettings);
```
Comment (soosinha) [inline server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java:375]: This is not required instead it should be added in settings during setup

**Comment by shwetathareja** [comment]
@shiv0408 please rebase from main.

**Review by shwetathareja — APPROVED** [review]

**Comment by shiv0408** [comment]
> ❌ Gradle check result for https://github.com/opensearch-project/OpenSearch/commit/c2b1e3f03db87db2869bb2f38c47976c6e1f97e2: [FAILURE](https://build.ci.opensearch.org/job/gradle-check/47318/)

[org.opensearch.search.approximate.ApproximatePointRangeQueryTests.testApproximateRangeShortCircuitAscSort](https://build.ci.opensearch.org/job/gradle-check/47318/testReport/junit/org.opensearch.search.approximate/ApproximatePointRangeQueryTests/testApproximateRangeShortCircuitAscSort/) - #15600

**Review by shwetathareja — APPROVED** [review]

**Pull request merged**